src/bindings/c/include/openvino/c/ov_shape.h

```diff
@@ -0,0 +1,22 @@
+// Copyright (C) 2018-2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+/**
+ * @brief This is a header file for ov_shape C API
+ *
+ * @file ov_shape.h
+ */
+
+#pragma once
+
+#define MAX_DIMENSION 8
+
+/**
+ * @struct ov_shape_t
+ * @brief Reprents a static shape.
+ */
+typedef struct {
+    int64_t rank;
+    int64_t dims[MAX_DIMENSION];
```

Comment:
It should be a very simple and straightforward structure, is they any possible scaling? Simple is beauty, why put so simple structure into a complex usage?

Reply:
We shouldn't introduce synthetic limits, Shape and partial shape can have any number of dimension in C++. In C API partial shape also can have any number of dimensions.
So why in this case we introduce synthetic limits for Shape in C? What if we would have a case with rang > 8?

Reply:
In very many C driver or application, we always use such method to avoid lots trouble.
Yes, it will bring some limitation, I admit, but it very low probability to hit such limitation and it will bring huge convenience to user. I think we can benefit much more than its limitation.

Reply:
The approach with pointer has only one difference between the current approach, we need to call `alloc` at the beginning and at the end call `free`.
But it relax all limitations.

Reply:
done